From 7f66b026bbdda6bd3ddf7f617bf2c3a475e4a665 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 1 Jun 2018 19:04:37 +0200
Subject: [PATCH 1/3] core: when we can't enqueue OnFailure= job show full
 error message
Bug: https://github.com/systemd/systemd/issues/8398
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1849261
Origin: upstream, https://github.com/systemd/systemd/pull/9158

Let's ask for the full error message and show it, there's really no
reason to just show the crappy errno error.
---
 src/core/unit.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -2151,6 +2151,7 @@
         Unit *other;
         Iterator i;
         void *v;
+        int r;
 
         assert(u);
 
@@ -2160,11 +2161,11 @@
         log_unit_info(u, "Triggering OnFailure= dependencies.");
 
         HASHMAP_FOREACH_KEY(v, other, u->dependencies[UNIT_ON_FAILURE], i) {
-                int r;
+                _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
 
-                r = manager_add_job(u->manager, JOB_START, other, u->on_failure_job_mode, NULL, NULL);
+                r = manager_add_job(u->manager, JOB_START, other, u->on_failure_job_mode, &error, NULL);
                 if (r < 0)
-                        log_unit_error_errno(u, r, "Failed to enqueue OnFailure= job: %m");
+                        log_unit_warning_errno(u, r, "Failed to enqueue OnFailure= job, ignoring: %s", bus_error_message(&error, r));
         }
 }
 
